src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -176,6 +178,42 @@ public static boolean itinerariesMatch(Itinerary referenceItinerary, Itinerary c
         return true;
     }
 
+    /**
+     * Checks that the specified itinerary is on the same day as the specified date/time.
+     * @param itinerary the itinerary to check.
+     * @param requestDateTime the request date/time to check, in the OTP's time zone.
+     * @param tripIsArriveBy true to check the itinerary endtime, false to check the startTime.
+     * @return true if the itinerary's startTime or endTime is one the same day as the day of the specified date and time.
+     */
+    public static boolean isSameDay(Itinerary itinerary, ZonedDateTime requestDateTime, boolean tripIsArriveBy) {
```

Comment:
I'd suggest changing the `requestDateTime` arg name to `date`, `isSameDay` to `occursOnDate`, and `tripIsArriveBy` to `arriveBy`. Also, just a thought: instead of passing around the arriveBy boolean, should we just add it as a field on `Itinerary`? cc @evansiroky

Reply:
Partially addressed in #112.